File: pluto/fig_1_11.jl
### A Pluto.jl notebook ###
# v0.19.0

using PlutoUI
using WGLMakie
include("common/hydrogenic.jl")
using .Hydrogenic

md"""# Figure 1.11
Radial distribution P(r) for 1s; r_max = a0/Z."""

@bind Z Slider(1:5, default=1)

xs = range(0, 8; length=400)
rs = [x*a0 for x in xs]
P = [radial_distribution(1,0,r; Z=Z) for r in rs]

rmax = a0/Z

fig = Figure(resolution=(800,400))
ax = Axis(fig[1,1], xlabel="r/a0", ylabel="P(r)")
lines!(ax, xs, P, color=:steelblue)
vline!(ax, rmax/a0, color=:orange, linestyle=:dash)
fig

File: web/src/figures/fig_1_11.ts
import Plotly from 'plotly.js-dist-min';
import { radialDistribution, constants } from '../common/hydrogenic';
import { slider } from '../common/ui';

const controls = document.getElementById('controls')!;
const plot = document.getElementById('plot')!;
const caption = document.getElementById('caption')!;
caption.textContent = 'FIGURE 1.11 Radial distribution P(r) for 1s; r_max = a₀/Z.';

const zSlider = slider(controls, 'Z', 1, 5, 1);

const compute = () => {
  const Z = parseInt(zSlider.value, 10);
  const xs: number[] = [];
  const P: number[] = [];
  for (let i = 0; i <= 300; i++) {
    const r = (8 * constants.a0 * i) / 300;
    xs.push(r / constants.a0);
    P.push(radialDistribution(1, 0, r, Z));
  }
  const rmax = constants.a0 / Z;
  Plotly.newPlot(plot, [
    { x: xs, y: P, mode: 'lines', name: 'P(r)', line: { color: '#1f6f8b' } },
    { x: [rmax / constants.a0, rmax / constants.a0], y: [0, Math.max(...P)], mode: 'lines', line: { dash: 'dot', color: '#e67e22' }, name: 'r_max' }
  ], {
    margin: { l: 60, r: 20, t: 20, b: 40 },
    xaxis: { title: 'r/a₀' },
    yaxis: { title: 'P(r)' },
    legend: { orientation: 'h' }
  }, { displayModeBar: false });
};

zSlider.addEventListener('input', compute);
compute();
